### PR TITLE
Fix/attribute metadata

### PR DESCRIPTION
**Pull request by IvanHdzC**
This PR fixes the mentioned exception on. #1902. 

On the entity that failed when persisting on the comment, this would be the persistance json.

Persisting data at NGSICKANSink (orgName=dipbad, pkgName=dipbad_envairqualityrespira, resName=bada1_respira, data={"recvTime":"2020-09-04T07:05:54.756Z","fiwareServicePath":"/envAirQualityRESPIRA","entityId":"bada1","entityType":"RESPIRA","TimeInstant":"2020-07-01T20:32:06.000Z","TimeInstant_md":[{"name":"dateCreated","type":"DateTime","value":"2020-01-10T10:08:33.000Z"},{"name":"dateModified","type":"DateTime","value":"2020-07-08T10:55:38.000Z"}],"airQualityIndex":31,"airQualityIndex_md":[{"name":"dateCreated","type":"DateTime","value":"2020-01-10T10:08:33.000Z"},{"name":"dateModified","type":"DateTime","value":"2020-07-08T10:55:38.000Z"}]})

### CODE REVIEW

**Review by AlvaroVega — APPROVED** [review]
LGTM

**Comment by AlvaroVega** [comment]
This PR reopens https://github.com/telefonicaid/fiware-cygnus/pull/1919 so no CNR update is needed

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/aggregation/NGSIGenericColumnAggregator.java [hunk 1]
```diff
@@ -108,7 +108,8 @@ public void aggregate(NGSIEvent event) {
             String attrType = contextAttribute.getType();
             JsonElement attrValue = contextAttribute.getValue();
             String attrMetadata = contextAttribute.getContextMetadata();
-            JsonArray jsonAttrMetadata = new Gson().fromJson(attrMetadata, JsonArray.class);
+            JsonParser jsonParser = new JsonParser();
+            JsonArray jsonAttrMetadata = (JsonArray) jsonParser.parse(attrMetadata);
```
Comment (fgalan) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/aggregation/NGSIGenericColumnAggregator.java:112]: Maybe some test should be added covering the failing case (now solved by the PR)?

Comment (AlvaroVega) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/aggregation/NGSIGenericColumnAggregator.java:112]: previous PR https://github.com/telefonicaid/fiware-cygnus/pull/1919 were modifying existing tests about this feature

Comment (fgalan) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/aggregation/NGSIGenericColumnAggregator.java:112]: So I understand new tests are not needed.

In that case, NTC.

**Review by fgalan — APPROVED** [review]
LGTM

File: cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/aggregation/NGSIGenericRowAggregator.java [hunk 2]
```diff
@@ -92,7 +96,8 @@ public void aggregate(NGSIEvent event) {
             aggregation.get(NGSIConstants.ATTR_NAME).add(new JsonPrimitive(attrName));
             aggregation.get(NGSIConstants.ATTR_TYPE).add(new JsonPrimitive(attrType));
             aggregation.get(NGSIConstants.ATTR_VALUE).add(attrValue);
-            aggregation.get(NGSIConstants.ATTR_MD).add(new JsonPrimitive(attrMetadata));
+            aggregation.get(NGSIConstants.ATTR_MD).add(jsonAttrMetadata);
+            //aggregation.get(NGSIConstants.ATTR_MD).add(new JsonPrimitive(attrMetadata));
```
Comment (AlvaroVega) [inline cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/aggregation/NGSIGenericRowAggregator.java:100]: comment legacy?

Comment (IvanHdzC) [reply cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/aggregation/NGSIGenericRowAggregator.java]: Fixed in [611a63c](https://github.com/telefonicaid/fiware-cygnus/pull/1923/commits/611a63ca19c1bcebc88e8d2bca4119031679c8b0)

**Review by AlvaroVega — APPROVED** [review]
LGTM

**Pull request merged**